### ISSUE TITLE
Where is the reported data processing?

### ISSUE THREAD

**Issue by ichitaka** [Pearl-UTexas/EMPATHIC#5]
It is reported, that processing in the form of a running average with the head pose data is applied, and Fourier transformation coefficients are calculated. I can't find these transformations anywhere in this repository.

**Comment by yuchen93** [comment]
Thanks for the question. We apologize for not organizing the repo in the most straight-forward manner. The data processing logic can be found in the feature extractor for the online learning agent, which is part of the Robotaxi repo. More specifically in this function: https://github.com/Pearl-UTexas/RobotaxiEnv/blob/59fe65ee9db2da979dec62f3cb05407a3e4d73c1/robotaxi/detectors/feedback_detector.py#L89

**Comment by ichitaka** [comment]
@yuchen93 Thank you very much! This has been quite helpful. I still can't understand why your processed data has 3 data points per game state.

So when I look into '2iA4jV97rl_episode_30_detectionFrames_0.csv' for example, each value of the first column is repeated 3 times. From my understanding, this column is the mapping to game states.

**Comment by yuchen93** [comment]
This is because the video is recorded at different frame rate than the actual game steps. This process is explained in our paper and also plotted in Figure 5. For each game frame, we average consecutive feature predictions and turn the original 30FPS video recording into 6FPS and the actual game run at 1.2FPS, which will produce 3 prediction frames per game state.

**Comment by ichitaka** [comment]
But this is exactly what I mean. In 5 seconds, there will be 6 game states and 30 aggregated frames. Now, when I divide this through 6 to get the number for a single game state, I will have 5 aggregated frames per game state.

Also, you mean max pooling instead of average, right?

**Comment by yuchen93** [comment]
Yes, thanks for the question. Now I understand why you are confused. For the aggregated frames in the collected data, the actual recorded video frames do not have consistent FPS and 30 was the max. Most game frames have a range of 17-25 frames and for consistency we computed 3 aggregated frames per game frame. For online learning we just used every 5 consecutive image frames. Our prediction window is large enough to cover multiple game states and therefore the model wasn’t super sensitive to this difference in the FPS of aggregated frames.

**Comment by ichitaka** [comment]
@yuchen93 Thank you very much, this makes far more sense now! One last question regarding the processing. The processed data has quite different ranges of values for the features in comparison to anything that I can get out of OpenFace. Like for the intensity action units (e.g. AU01_r) one would expect values between 0 and 5, but none of your features have such ranges. Most of the data has values up to 1. Is there any other kind of processing you applied, like normalization or scaling?

**Comment by yuchen93** [comment]
We clipped the regression outputs at one, specifically with this line: https://github.com/Pearl-UTexas/RobotaxiEnv/blob/59fe65ee9db2da979dec62f3cb05407a3e4d73c1/robotaxi/detectors/feedback_detector.py#L148

**Comment by ichitaka** [comment]
Thanks a lot. I still fail to reproduce the Kendall Tau values you report in your paper. I've basically added this to the test_facs.py loop.


    def model_metrics(test, pred):
        tau, p_value = stats.kendalltau(test, pred)
        print('Correlation:', round(tau, 2), ' p-value:', round(p_value, 2))
  
    preds_df = pd.DataFrame()

    # Testing the trained model on holdout set, and record the reward class probability for each pickup event.
    for i,  (img_features, labels, _, event_mask, _, _, _) in enumerate(test_loader):
        img_features = Variable(img_features.float()).to(device)
        event_mask = Variable(event_mask.float()).to(device)
        labels = Variable(labels).to(device)
        outputs, _ = net(img_features.float(), event_mask)
        outputs = outputs.cpu().data.float()                
        y_pred = torch.argmax(outputs, dim=1).numpy()
        batch_df = pd.DataFrame()
        batch_df['y_pred'] = pd.Series(y_pred)
        batch_df['labels'] = pd.Series(labels)
        preds_df = preds_df.append(batch_df)

    model_metrics(preds_df['labels'], preds_df['y_pred'])
